oak_functions/lookup/src/lib.rs

```diff
@@ -281,35 +310,99 @@ mod tests {
+        assert_eq!(lookup_data.len(), 2);
+    }
+
+    #[test]
+    fn test_update_lookup_data_two_chunks() {
+        let manager = LookupDataManager::new_empty(TestLogger {});
+        let lookup_data_0 = manager.create_lookup_data();
+
+        let update_status = manager.update_start(create_test_data(0, 2));
+        assert_eq!(update_status, UpdateStatus::Started);
+        let lookup_data_1 = manager.create_lookup_data();
+
+        let update_status = manager.update_finish(create_test_data(2, 4));
+        assert_eq!(update_status, UpdateStatus::Finished);
+        let lookup_data_2 = manager.create_lookup_data();
+
+        assert_eq!(lookup_data_0.len(), 0);
+        assert_eq!(lookup_data_1.len(), 0);
+        assert_eq!(lookup_data_2.len(), 4);
+    }
+
+    #[test]
+    fn test_update_lookup_data_abort_after_second_start() {
+        let manager = LookupDataManager::new_empty(TestLogger {});
+        let lookup_data_0 = manager.create_lookup_data();
+
+        let update_status = manager.update_start(create_test_data(0, 2));
+        assert_eq!(update_status, UpdateStatus::Started);
+        let lookup_data_1 = manager.create_lookup_data();
+
+        let update_status = manager.update_start(create_test_data(2, 4));
+        assert_eq!(update_status, UpdateStatus::Aborted);
+        let lookup_data_2 = manager.create_lookup_data();
+
+        assert_eq!(lookup_data_0.len(), 0);
+        assert_eq!(lookup_data_1.len(), 0);
+        assert_eq!(lookup_data_2.len(), 0);
+    }
+
+    #[test]
+    fn test_update_lookup_data_abort_after_second_start_and_finish() {
+        let manager = LookupDataManager::new_empty(TestLogger {});
+        let lookup_data_0 = manager.create_lookup_data();
+
+        let update_status = manager.update_start(create_test_data(0, 2));
+        assert_eq!(update_status, UpdateStatus::Started);
+        let lookup_data_1 = manager.create_lookup_data();
+
+        let update_status = manager.update_start_and_finish(create_test_data(2, 4));
+        assert_eq!(update_status, UpdateStatus::Aborted);
+        let lookup_data_2 = manager.create_lookup_data();
+
+        assert_eq!(lookup_data_0.len(), 0);
+        assert_eq!(lookup_data_1.len(), 0);
+        assert_eq!(lookup_data_2.len(), 0);
+    }
+
     #[test]
     fn test_format_bytes() {
         // Valid UTF-8 string.
         assert_eq!("🚀oak⭐", format_bytes("🚀oak⭐".as_bytes()));
         // Incorrect UTF-8 bytes, as per https://doc.rust-lang.org/std/string/struct.String.html#examples-3.
         assert_eq!("[0, 159, 146, 150]", format_bytes(&[0, 159, 146, 150]));
     }
+
+    // Create test data with size distinct keys between inclusive start and exclusive end.
+    fn create_test_data(start: i32, end: i32) -> Data {
+        let mut map = HashMap::new();
+        for i in start..end {
+            map.insert(
+                format!("key{}", i).into_bytes(),
+                format!("value{}", i).into_bytes(),
+            );
+        }
```

Comment:
could probably be something more functional, like `(start..end).map(|i| (format!("key{}", i).into_bytes(), format!("value{}", i).into_bytes())).collect()`

Reply:
Done.